From 00de1a6f9ff8111dfc2f3c11eb8a070a8863484a Mon Sep 17 00:00:00 2001
From: gvwifi build <build@local>
Date: Sat, 3 Oct 2026 11:19:46 -0400
Subject: [PATCH 1/3] recovery: build with crypto on android-14.1 trees

- keystore AIDL NDK libraries are '-ndk' on Android 13+ (apc, authorization,
  maintenance, keystore2-V1); the '-ndk_platform' names no longer exist
- libtar links libvold, which now also needs libboot_control_client,
  libsysutils (NetlinkEvent) and libasync_safe

Change-Id: I632abafee94cd06c6769960de8df390007e9dec9
---
 Android.mk        |  6 +++---
 libtar/Android.mk | 12 +++++++-----
 2 files changed, 10 insertions(+), 8 deletions(-)

diff --git a/Android.mk b/Android.mk
index 78b94308..ff9296c8 100755
--- a/Android.mk
+++ b/Android.mk
@@ -334,14 +334,14 @@ ifeq ($(TW_INCLUDE_CRYPTO), true)
     TW_INCLUDE_CRYPTO_FBE := true
     LOCAL_CFLAGS += -DTW_INCLUDE_FBE
     LOCAL_SHARED_LIBRARIES += android.frameworks.stats@1.0 android.hardware.authsecret@1.0 \
-	android.security.authorization-ndk_platform \
+	android.security.authorization-ndk \
         android.hardware.oemlock@1.0 libf2fs_sparseblock \
         libandroidicu.recovery \
         android.hardware.gatekeeper@1.0 \
         android.hardware.weaver@1.0 \
         android.frameworks.stats@1.0 \
-        android.security.maintenance-ndk_platform \
-        android.system.keystore2-V1-ndk_platform \
+        android.security.maintenance-ndk \
+        android.system.keystore2-V1-ndk \
         libkeyutils \
         liblog \
         libsqlite.recovery \
diff --git a/libtar/Android.mk b/libtar/Android.mk
index c3bb691f..4f2cd1fe 100755
--- a/libtar/Android.mk
+++ b/libtar/Android.mk
@@ -14,7 +14,9 @@ LOCAL_C_INCLUDES += external/libselinux/include
 LOCAL_SHARED_LIBRARIES += libselinux
 
 ifeq ($(TW_INCLUDE_CRYPTO_FBE), true)
-    LOCAL_STATIC_LIBRARIES += libvold libscrypt_static
+    LOCAL_STATIC_LIBRARIES += libvold libscrypt_static libasync_safe
+    # libvold (android-14.1) also needs the boot control client and NetlinkEvent
+    LOCAL_SHARED_LIBRARIES += libboot_control_client libsysutils
     LOCAL_SHARED_LIBRARIES +=  \
         android.hardware.boot@1.0 \
         android.hardware.confirmationui@1.0 \
@@ -23,10 +25,10 @@ ifeq ($(TW_INCLUDE_CRYPTO_FBE), true)
         android.hardware.keymaster@4.0 \
         android.hardware.keymaster@4.1 \
         android.hardware.weaver@1.0 \
-        android.security.apc-ndk_platform \
-        android.system.keystore2-V1-ndk_platform \
-        android.security.authorization-ndk_platform \
-        android.security.maintenance-ndk_platform \
+        android.security.apc-ndk \
+        android.system.keystore2-V1-ndk \
+        android.security.authorization-ndk \
+        android.security.maintenance-ndk \
         libselinux \
         libbinder_ndk \
         libext4_utils \
